core/src/mender-client.c

```diff
@@ -1080,11 +1080,27 @@ mender_client_update_work_function(void) {
     /* reset the currently used update module */
     mender_update_module = NULL;
 
+    {
+        cJSON *deployment_id_j = cJSON_GetObjectItem(mender_client_deployment_data, "id");
+        if (NULL != deployment_id_j) {
+            deployment_id = deployment_id_j->valuestring;
+        }
```

Comment:
Pre-existing,

What happens if there is no deployment id? Can we continue at all? (We won't be able to send status to the server, for example)